packages/react-server-cli/src/compileClient.js

```diff
@@ -165,6 +166,13 @@ const packageCodeForBrowser = (entrypoints, outputDir, outputUrl, hot, minify, l
 				},
 			],
 		},
+		resolve: {
+			alias: {
+				// These need to be singletons.
+				"react"        : callerDependency("react"),
```

Comment:
Not if you're doing local dev with `npm link` -- npm can't/doesn't/won't (one of those, not sure which) flatten the dependency tree across multiple modules.